File: MATTHEW/leetcode/AIcoding/2.Gemini-2.0/cpp_solutions/3433-count-mentions-per-user.cpp
#include <vector>
#include <string>
#include <sstream>
#include <unordered_map>

using namespace std;

struct Users {
    int user_id;
    string name;
};

struct Tweets {
    int tweet_id;
    int poster_id;
    string tweet;
};

vector<vector<string>> countMentions(vector<Users>& users, vector<Tweets>& tweets) {
    unordered_map<int, string> userMap;
    for (auto& user : users) {
        userMap[user.user_id] = user.name;
    }

    unordered_map<string, int> mentionCount;
    for (auto& tweet : tweets) {
        stringstream ss(tweet.tweet);
        string word;
        while (getline(ss, word, ' ')) {
            if (word.length() > 0 && word[0] == '@') {
                string mention = word.substr(1);
                mentionCount[mention]++;
            }
        }
    }

    vector<vector<string>> result;
    for (auto& pair : mentionCount) {
        result.push_back({pair.first, to_string(pair.second)});
    }

    return result;
}

File: MATTHEW/leetcode/AIcoding/3.DeepSeek-v3/javascript_solutions/3433-count-mentions-per-user.js
var countMentions = function(tweets) {
    const mentionCount = {};

    for (const tweet of tweets) {
        const mentions = tweet.match(/@(\w+)/g) || [];
        const mentionedUsers = new Set();

        for (const mention of mentions) {
            const username = mention.substring(1);
            mentionedUsers.add(username);
        }

        for (const user of mentionedUsers) {
            if (!mentionCount[user]) {
                mentionCount[user] = 0;
            }
            mentionCount[user]++;
        }
    }

    return mentionCount;
};